src/routes/users.rs

```diff
@@ -54,6 +92,160 @@ async fn post_login(
     Ok(Redirect::to(&redirect_to).into_response())
 }
 
+async fn get_login_google(
+    State(state): State<AppState>,
+    session: Session,
+) -> ResponseResult<Response> {
+
+    let google_client_id = ClientId::new(
+        state.oauth_google_client_id
+    );
+    let google_client_secret = ClientSecret::new(
+        state.oauth_google_client_secret
+    );
+    let issuer_url = 
+        IssuerUrl::new("https://accounts.google.com".to_string()).context("failed to reach issuer")?;
+
+    let provider_metadata = task::spawn_blocking(move || {
```

Comment:
You can use `discover_async` to eliminate the call to `tokio::spawn_blocking`.

Reply:
I'm so sorry but I couldn't find any docs regarding `discover_async`, could you please share one so that I can dive into the docs? Thanks in advance

Reply:
Sure! [This is the function I was talking about](https://docs.rs/openidconnect/3.5.0/openidconnect/struct.ProviderMetadata.html#method.discover_async).